### ISSUE TITLE
Fatal error when running kubectl after enabling Kubernetes

### ISSUE THREAD

**Issue by danfinn** [docker/for-mac#4447]
Diagnostic ID : 29C24300-9E36-4FFB-99AA-D689E229C1D8/20200413224808

I finally got Kubernetes to start after struggling with that for a bit.  I ended up having to fully uninstall and re-install.  It now shows that it's running and has docker-desktop set as context.

Unfortunately when I attempt to run a basic kubectl command, such as kubectl version I get the following error when run as my normal user:

>kubectl version
failed MSpanList_Insert 0x1c82000 0x2d7e918ebe 0x0
fatal error: MSpanList_Insert
runtime stack:
runtime.throw(0x1a22f8b)
	/usr/src/go/src/runtime/panic.go:491 +0xad fp=0x7ffeefbff4b0 sp=0x7ffeefbff480
runtime.MSpanList_Insert(0x1a59a08, 0x1c82000)
	/usr/src/go/src/runtime/mheap.c:692 +0x8f fp=0x7ffeefbff4d8 sp=0x7ffeefbff4b0
MHeap_FreeSpanLocked(0x1a56600, 0x1c82000, 0x100)
	/usr/src/go/src/runtime/mheap.c:583 +0x163 fp=0x7ffeefbff518 sp=0x7ffeefbff4d8
MHeap_Grow(0x1a56600, 0x8, 0x0)
	/usr/src/go/src/runtime/mheap.c:420 +0x1a8 fp=0x7ffeefbff558 sp=0x7ffeefbff518
MHeap_AllocSpanLocked(0x1a56600, 0x1, 0x0)
	/usr/src/go/src/runtime/mheap.c:298 +0x365 fp=0x7ffeefbff598 sp=0x7ffeefbff558
mheap_alloc(0x1a56600, 0x1, 0x12, 0x0)
	/usr/src/go/src/runtime/mheap.c:190 +0x121 fp=0x7ffeefbff5c0 sp=0x7ffeefbff598
runtime.MHeap_Alloc(0x1a56600, 0x1, 0x10000000012, 0xcf29)
	/usr/src/go/src/runtime/mheap.c:240 +0x66 fp=0x7ffeefbff5f8 sp=0x7ffeefbff5c0
MCentral_Grow(0x1a5e378, 0x0)
	/usr/src/go/src/runtime/mcentral.c:197 +0x8b fp=0x7ffeefbff660 sp=0x7ffeefbff5f8
runtime.MCentral_CacheSpan(0x1a5e378, 0x0)
	/usr/src/go/src/runtime/mcentral.c:85 +0x167 fp=0x7ffeefbff698 sp=0x7ffeefbff660
runtime.MCache_Refill(0x1c7e000, 0x12, 0x0)
	/usr/src/go/src/runtime/mcache.c:90 +0xa0 fp=0x7ffeefbff6c0 sp=0x7ffeefbff698
runtime.mcacheRefill_m()
	/usr/src/go/src/runtime/malloc.c:368 +0x57 fp=0x7ffeefbff6e0 sp=0x7ffeefbff6c0
runtime.onM(0x153df88)
	/usr/src/go/src/runtime/asm_amd64.s:273 +0x9a fp=0x7ffeefbff6e8 sp=0x7ffeefbff6e0
runtime.mallocgc(0x120, 0x11b5fa0, 0x0, 0x0)
	/usr/src/go/src/runtime/malloc.go:178 +0x849 fp=0x7ffeefbff798 sp=0x7ffeefbff6e8
runtime.newobject(0x11b5fa0, 0x1c7e000)
	/usr/src/go/src/runtime/malloc.go:353 +0x49 fp=0x7ffeefbff7c0 sp=0x7ffeefbff798
runtime.newG(0x26dda)
	/usr/src/go/src/runtime/proc.go:233 +0x2a fp=0x7ffeefbff7d8 sp=0x7ffeefbff7c0
allocg(0x1a3f5a0)
	/usr/src/go/src/runtime/proc.c:925 +0x1f fp=0x7ffeefbff7e8 sp=0x7ffeefbff7d8
runtime.malg(0x8000, 0x1a3faa0)
	/usr/src/go/src/runtime/proc.c:2106 +0x1f fp=0x7ffeefbff818 sp=0x7ffeefbff7e8
runtime.mpreinit(0x1a40440)
	/usr/src/go/src/runtime/os_darwin.c:137 +0x27 fp=0x7ffeefbff830 sp=0x7ffeefbff818
mcommoninit(0x1a40440)
	/usr/src/go/src/runtime/proc.c:201 +0xc9 fp=0x7ffeefbff858 sp=0x7ffeefbff830
runtime.schedinit()
	/usr/src/go/src/runtime/proc.c:138 +0x55 fp=0x7ffeefbff880 sp=0x7ffeefbff858
runtime.rt0_go(0x7ffeefbff8b0, 0x2, 0x7ffeefbff8b0, 0x0, 0x2, 0x7ffeefbffa38, 0x7ffeefbffa40, 0x0, 0x7ffeefbffa48, 0x7ffeefbffa5f, ...)
	/usr/src/go/src/runtime/asm_amd64.s:95 +0x116 fp=0x7ffeefbff888 sp=0x7ffeefbff880

And if I run the same command as root (I'm not sure if I need to run it as root or not but I tried and noticed I get a different error):

`
kubectl version
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.5", GitCommit:"20c265fef0741dd71a66480e35bd69f18351daea", GitTreeState:"clean", BuildDate:"2019-10-15T19:16:51Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"darwin/amd64"}
The connection to the server localhost:8080 was refused - did you specify the right host or port?
`

If something is supposed to be listening on port 8080 it doesn't seem to be:

curl localhost:8080
curl: (7) Failed to connect to localhost port 8080: Connection refused

I don't see anything like this mentioned in the troubleshooting guide, not sure what else to do or what I might be missing.

**Comment by danfinn** [comment]
I was able to get kube commands working as root.  I noticed my root user had no .kube directory so I pointed it at my normal users .kube directory like so:

`
MACDFINN:.kube root# export KUBECONFIG=/Users/dfinn/.kube/config
MACDFINN:.kube root# kubectl version
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.5", GitCommit:"20c265fef0741dd71a66480e35bd69f18351daea", GitTreeState:"clean", BuildDate:"2019-10-15T19:16:51Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.5", GitCommit:"20c265fef0741dd71a66480e35bd69f18351daea", GitTreeState:"clean", BuildDate:"2019-10-15T19:07:57Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
MACDFINN:.kube root# kubectl get nodes
NAME             STATUS   ROLES    AGE   VERSION
docker-desktop   Ready    master   18h   v1.15.5
`

and as you can see I can now run kube commands as root.  I still cannot run them as my normal user though.

I'm starting to think this may have something to do with my personal use and Go but I'm not sure what is different about my user vs root.

**Comment by danfinn** [comment]
I looked at comparing env variables between the 2 users and looking for differences between root and my user.  I finally narrowed it down to the PATH env variable.  The contents weren't actually different but the order they were listed in was different.

I'm not sure why but this was causing an issue.  I adjusted the PATH to be the same between the 2 users and I can now run kube commands as my normal user.

>✔ {11:33} dfinn@MACDFINN:~> kubectl version
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.5", GitCommit:"20c265fef0741dd71a66480e35bd69f18351daea", GitTreeState:"clean", BuildDate:"2019-10-15T19:16:51Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.5", GitCommit:"20c265fef0741dd71a66480e35bd69f18351daea", GitTreeState:"clean", BuildDate:"2019-10-15T19:07:57Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
